addons/mass_mailing_event/models/event_registration.py

```diff
@@ -9,4 +9,8 @@ class EventRegistration(models.Model):
     _mailing_enabled = True
 
     def _mailing_get_default_domain(self, mailing):
-        return [('state', '!=', 'cancel')]
+        default_domain = [('state', 'not in', ['cancel', 'draft'])]
+        default_mailing_model_id = self.env.context.get('default_mailing_model_id')
+        if mailing.mailing_model_id.id == default_mailing_model_id:
+            return self.env.context.get('default_mailing_domain', default_domain)
```

Comment:
Seems like `_mailing_get_default_domain` should return a list of tuples, not a string, so the extracted value from the context should be processed through `ast.litteral_eval`

Also, personal preference, but I like this expression better for readability:
```suggestion
        if mailing.mailing_model_id.id == default_mailing_model_id and self.env.context.get('default_mailing_domain'):
            return ast.literal_eval(self.env.context['default_mailing_domain'])
```
But feel free to ignore my rambling :)